raiden-ts/src/transport/epics.ts

```diff
@@ -94,45 +94,52 @@ const userRe = /^@(0x[0-9a-f]{40})[.:]/i;
 export const initMatrixEpic = (
   {  }: Observable<RaidenAction>,
   state$: Observable<RaidenState>,
-  { address, network, signer, matrix$ }: RaidenEpicDeps,
+  { address, network, signer, matrix$, config$ }: RaidenEpicDeps,
 ): Observable<ActionType<typeof matrixSetup>> =>
   state$.pipe(
     first(),
     mergeMap(state => {
       const server: string | undefined = get(state, ['transport', 'matrix', 'server']),
         setup: RaidenMatrixSetup | undefined = get(state, ['transport', 'matrix', 'setup']);
-      if (server) {
-        // use server from state/settings
-        return of({ server, setup });
-      } else {
-        const knownServersUrl =
-          MATRIX_KNOWN_SERVERS_URL[network.name] || MATRIX_KNOWN_SERVERS_URL.default;
-        // fetch servers list and use the one with shortest http round trip time (rtt)
-        return from(fetch(knownServersUrl)).pipe(
-          mergeMap(response => {
-            if (!response.ok)
-              return throwError(
-                new Error(
-                  `Could not fetch server list from "${knownServersUrl}" => ${response.status}`,
-                ),
-              );
-            return response.text();
-          }),
-          mergeMap(text => from(yamlListToArray(text))),
-          mergeMap(server => matrixRTT(server)),
-          toArray(),
-          map(rtts => sortBy(rtts, ['rtt'])),
-          map(rtts => {
-            if (!rtts[0] || typeof rtts[0].rtt !== 'number' || isNaN(rtts[0].rtt))
-              throw new Error(`Could not contact any matrix servers: ${JSON.stringify(rtts)}`);
-            return rtts[0].server;
-          }),
-          map(server => ({
-            server: server.includes('://') ? server : `https://${server}`,
-            setup: undefined,
-          })),
-        );
-      }
+
+      return config$.pipe(
```

Comment:
Can we simplify this somehow, maybe through `combineLatest(state$, config$)` or something like that?

Reply:
`withLatestFrom` before the top-level `mergeMap`